packages/react-i18n/src/utils/i18n.utils.js

```diff
@@ -57,7 +58,16 @@ export const translate = (lang, i18nNames = {}, errorCallback = _noop, parseHTML
       translation = combineKey;
     }
 
-    const translatedResult = sprintf(translation, { ...data, ...i18nNames, number });
+    // escape special characters (% for now)
+    const translationValues = _mapValues({ ...data, ...i18nNames, number }, value => {
+      if (value) {
```

Comment:
According to the unit test, value can be a number. We should probably check that we have a string, what di you think? 
```suggestion
     if (typeof value === 'string') {
```